cognite/seismic/protos/types.proto

```diff
@@ -87,11 +87,14 @@ message SurveyGridTransformation {
     This is not valid for traces that are synthetically generated from interpolation.
  **/
 message Trace {
-    bytes trace_header = 1;
-    google.protobuf.Int32Value iline = 2; // The inline number. Within a 3D survey, a inline represents a seismic line parallel to the direction in which the data was adquired. 
-    google.protobuf.Int32Value xline = 3; // The xline number. Within a 3D survey, a xline represents a seismic line perpendicular to the direction in which the data was adquired. 
-    repeated float trace = 4;                // The underlying array of floats representing samples
-    Coordinate coordinate = 5;
+    bytes trace_header = 1;                             // The binary SEGY trace header, if requested.
+    google.protobuf.Int32Value iline = 2;               // The inline number. Only populated for 3D seismics.
+    google.protobuf.Int32Value xline = 3;               // The xline number. Only populated for 3D seismics.
+    google.protobuf.Int32Value cdp = 6;                 // The cdp number. Only populated for 2D seismics indexed by cdp.
+    google.protobuf.Int32Value shotpoint = 7;           // The cdp number. Only populated for 2D seismics indexed by shotpoint.
+    google.protobuf.Int32Value energy_source_point = 8; // The cdp number. Only populated for 2D seismics indexed by energy source point.
```

Comment:
Could we just return all the values rather than only returning values that have been indexed? We're just reading a few bytes from the trace header in both tiers anyways so it's not like its super costly

Reply:
In a future trace store implementation with trace data stored in compressed blocks we may not actually have header values other than the header keys on hand, so returning all values may require a detour to some other storage method for headers. This _may_ be a performance liability, so I'm inclined to omit other headers for now rather than providing them and implicitly promising to  always provde them. It's always easier to add these later than to remove them after someone's become dependent on their presence.